swim/join_sender.go

```diff
@@ -65,6 +65,10 @@ type joinOpts struct {
 	size              int
 	maxJoinDuration   time.Duration
 	parallelismFactor int
+
+	// hostProvider is the HostProvider that this joinSender will use to
+	// enumerate bootstrap hosts.
+	hostProvider HostProvider
```

Comment:
I am thinking this should be renamed to `discoverProvider` / `DiscoverProvider`.